### PR TITLE
[DDCE-2055]:charities-registration-frontend Play 2.8 upgrade

### PR DESCRIPTION
**Pull request by anburoach**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -6,33 +6,35 @@ object AppDependencies {
 
   val compile = Seq(
     play.sbt.PlayImport.ws,
-    "org.reactivemongo"         %% "play2-reactivemongo"             % "0.20.11-play27",
+    "org.reactivemongo"         %% "play2-reactivemongo"             % "0.20.11-play28",
     "org.reactivemongo"         %% "reactivemongo-bson-api"          % "0.20.11",
     "com.typesafe.play"         %% "play-iteratees"                  % "2.6.1",
     "com.typesafe.play"         %% "play-iteratees-reactive-streams" % "2.6.1",
```
Comment (JMaio) [inline project/AppDependencies.scala:12]: Leaving a note here regarding the `play-iteratees` package, maybe this needs to be looked into in a separate issue:
> https://github.com/playframework/play-iteratees
> End of Life
> The active Playframework contributors consider this repository has reached End of Life and archived it.
> This repository is not being used anymore and won't get any further updates.

File: test/controllers/RegistrationSentControllerSpec.scala [hunk 2]
```diff
@@ -91,24 +95,24 @@ class RegistrationSentControllerSpec extends SpecBase with ImplicitDateFormatter
...
         .flatMap(_.set(ApplicationSubmissionDatePage, inject[TimeMachine].now()))
         .success.value)))
 
       val result = controller.onPageLoad()(fakeRequest)
 
       status(result) mustEqual OK
-      contentAsString(result) mustEqual view(dayToString(inject[TimeMachine].now().plusDays(28)),
+      contentAsString(result) mustEqual view(dayToString(inject[TimeMachine].now().plusDays(daysToAdd)),
         dayToString(inject[TimeMachine].now(), dayOfWeek = false),
         "123456789", emailOrPost = false, noEmailOrPost = true,
```
Comment (JMaio) [inline test/controllers/RegistrationSentControllerSpec.scala:117]: stray `acknowledgementReferenceNo`?

Comment (anburoach) [reply test/controllers/RegistrationSentControllerSpec.scala:117]: this is a dummy acknowledgeReferenceNo for test, so no issues

File: test/navigation/NomineesNavigatorSpec.scala [hunk 3]
```diff
@@ -17,14 +17,14 @@
...
 import java.time.LocalDate
-
 import base.SpecBase
 import controllers.addressLookup.{routes => addressLookupRoutes}
 import controllers.nominees.{routes => nomineesRoutes}
 import controllers.routes
 import models._
 import models.addressLookup.{AddressModel, CountryModel}
 import models.nominees.OrganisationNomineeContactDetails
+import org.mockito.ArgumentMatchers.any
```
Comment (JMaio) [inline test/navigation/NomineesNavigatorSpec.scala:27]: is this matcher used anywhere? running the tests without it worked ok.

**Pull request merged**